packages/activity/src/index.ts

```diff
@@ -30,7 +30,22 @@ import { AsyncLocalStorage } from 'async_hooks';
 import { AbortSignal } from 'abort-controller';
 import { msToNumber } from '@temporalio/common';
 
-export { CancelledFailure } from '@temporalio/common';
+export { ActivityFunction, ActivityInterface, CancelledFailure } from '@temporalio/common';
+
+/**
+ * Throw this error from an Activity in order to make the Worker
+ * forget about this Activity.
+ *
+ * The Activity can be completed asynchronously using
+ * {@link AsyncCompletionClient}
+ */
+export class CompleteAsyncError extends Error {
+  public readonly name: string = 'CompleteAsyncError';
+
+  constructor() {
+    super();
+  }
+}
```

Comment:
Why not make this a return value rather than an exception? Nothing went "wrong" so it seems a return value would be better representing intent here.

I can see maybe not wanting to b/c it clutters the type signature of activities, but Typescript makes that so easy I'm not sure it's a big deal

Reply:
I actually started by making this a symbol and not an exception but typescript complains if you return it and I wouldn't want the activity to have to define a special return type because we use the activity return type in the workflow.

Reply:
Could introduce an intermediate type which is used internally and then narrowed to the user's type by the time it is returning to the workflow? I can see that being annoying though